src/js/components/AppListComponent.jsx

```diff
@@ -70,13 +70,19 @@ var AppListComponent = React.createClass({
     var sortKey = state.sortKey;
     var props = this.props;
 
-    return lazy(state.apps)
+    var appsSequence = lazy(state.apps);
+
+    if (props.filterText != null && props.filterText != "") {
+      appsSequence = appsSequence
+        .filter(function (app) {
+          return app.id.indexOf(props.filterText) !== -1;
```

Comment:
This has no advantage, because you don't use any lazy evaluation by using this without a sequence. Should be indeed slower than, because of more overhead.